test/ttmlir/Dialect/TTNN/reshape/reshape_folding_test.mlir

```diff
@@ -0,0 +1,12 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s| FileCheck %s
+#any_device_tile = #tt.operand_constraint<dram|l1|tile|any_device_tile>
+// Tests if we fold when translating from "ttir.reshape" which is called on the two same shapes.
+module @reshape_test {
+  func.func @main(%arg0: tensor<1xi32>) -> (tensor<1xi32> {jax.result_info = ""}) {
+    %0 = tensor.empty() : tensor<1xi32>
+    %1 = "ttir.reshape"(%arg0, %0) <{operand_constraints = [#any_device_tile, #any_device_tile], shape = [1 : i32]}> : (tensor<1xi32>, tensor<1xi32>) -> tensor<1xi32>
+    // CHECK: return %arg0 : tensor<1xi32, #{{.*}}>
+    // CHECK-NOT: %[[C:.*]] = "ttnn.reshape"[C:.*]]
```

Comment:
This will check that `ttnn.reshape` is not present after previous `CHECK` statement (which is checking `return` statement). Please move it above line 8. So FileCheck verify it before the `return` statement and not afterwards.